app/components/base/advanced-filter/list-filter-control.ts

```diff
@@ -0,0 +1,141 @@
+import { Component, Input } from "@angular/core";
+import { FormControl, FormGroup } from "@angular/forms";
+
+import { Filter, and, none, or, prop } from "app/utils/filter-builder";
+import { AdvancedFilter } from "./advanced-filter";
+import { AdvancedFilterControlBase } from "./control-base";
+
+export enum ListFilterType {
+    Include,
+    Exclude,
+}
+
+export interface ListFilterControlConfig {
+    /**
+     * If set to true all entries will be converted to numbers or ignored.
+     */
+    number?: boolean;
+
+    /**
+     * Allow list to take range.
+     * Range need to be speparated by the rangeSeparator
+     */
+    allowRanges?: boolean;
+
+    /**
+     * @default "-"
+     */
+    rangeSeparator?: string;
+}
+
+export class ListFilterControl extends AdvancedFilterControlBase {
+    private _valueControl = new FormControl("");
+
+    constructor(label: string, private _config: ListFilterControlConfig = {}) {
+        super(label);
+        if (!this._config.rangeSeparator) {
+            this._config.rangeSeparator = "|";
+        }
+    }
+
+    public formGroup(): FormGroup {
+        return new FormGroup({
+            type: new FormControl(ListFilterType.Include),
+            value: this._valueControl,
+        });
+    }
+
+    public buildFilter(data: { value: string, type: ListFilterType }): Filter {
+        if (!data.value) {
+            return none();
+        }
+        const values = this.parseValue(data.value);
+        if (data.type === ListFilterType.Include) {
+            const ranges = values.ranges.map(x => {
+                return and(prop(this.name).ge(x[0]), prop(this.name).le(x[1]));
+            });
+            return or(...ranges, prop(this.name).inList(values.items));
+        } else {
+            const ranges = values.ranges.map(x => {
+                return or(prop(this.name).lt(x[0]), prop(this.name).gt(x[1]));
+            });
+            return and(...ranges, prop(this.name).notInList(values.items));
+        }
+    }
+
+    public parseValue(value: string): { items: any[], ranges: any[][] } {
+        if (!value) {
+            return { items: [], ranges: [] };
+        }
+        let values = value.replace(/\s/g, "").split(",").filter(x => x !== "");
```

Comment:
Great example of a place to add some docs. My guess is that you're removing whitespaces?